src/praisonai/praisonai/ui/_auth.py

```
@@ -0,0 +1,169 @@
+                f"  Or:  Set PRAISONAI_ALLOW_DEFAULT_CREDS=1 (unsafe, for demo only)"
+            )
+        
+        if is_default_creds:
+            if is_external:
+                logger.warning(
+                    f"⚠️  Using default admin/admin credentials on external interface {bind_host}. "
+                    f"This is unsafe for production."
+                )
+            else:
+                logger.warning(
+                    f"⚠️  Using default admin/admin credentials on localhost. "
```

Ruff (0.15.10)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)